com.hybris.hyeclipse.commons/src/com/hybris/hyeclipse/commons/testers/AbstractFilePropertyTester.java

```diff
@@ -27,35 +26,45 @@
 	 * @return true if selected file matches at extension, false otherwise.
 	 */
 	protected boolean testSelectedFileByExtension(final Object receiver, final String fileExtension) {
-		return testSelectedFileByExtensions(receiver, new HashSet<>( Arrays.asList(fileExtension) ));
+		return testSelectedFilesByExtensions(receiver, new HashSet<>( Arrays.asList(fileExtension) ));
 	}
 	
 	/**
-	 * Check whether selected file matches set of extensions 
+	 * Check whether selected file(s) matches set of extensions 
 	 * 
-	 * @param receiver selected file
+	 * @param receiver selected file(s)
 	 * @param fileExtensions set of extensions to compare
-	 * @return true if selected file matches at least one extensions, false otherwise.
+	 * @return true if selected file(s) matches at least one extensions, false otherwise.
 	 */
-	protected boolean testSelectedFileByExtensions(final Object receiver, final Set<String> fileExtensions) {
+	@SuppressWarnings({"unchecked", "restriction"})
+	protected boolean testSelectedFilesByExtensions(final Object receiver, final Set<String> fileExtensions) {
+		boolean isValid = false;
+		
 		if (receiver instanceof Set) {
 			final Set<Object> set = (Set<Object>) receiver;
 			
 			if (set.size() == 1 && set.iterator().next() instanceof TextSelection) {
 				final IWorkbenchWindow window = PlatformUI.getWorkbench().getActiveWorkbenchWindow();
```

Comment:
I see this combination of getting current window and window on that file in many places. Could that be a seperate method on util class ? Like get current file or something  ?